adal/src/androidTest/java/com/microsoft/aad/adal/AuthenticationActivityUnitTest.java

```diff
@@ -216,41 +223,54 @@ public void testWebviewAuthenticationException() throws IllegalArgumentException
     }
 
 
-    @SmallTest
-    @UiThreadTest
-    public void testWebviewSslprotectedredirectURL() throws IllegalArgumentException,
-            NoSuchFieldException, IllegalAccessException, InvocationTargetException,
-            ClassNotFoundException, NoSuchMethodException, InstantiationException,
-            InterruptedException, ExecutionException {
-        startActivity(mIntentToStartActivity, null, null);
-        mActivity = getActivity();
+    @Test
+    public void testWebviewSslprotectedredirectURL() throws Throwable {
+        mActivityRule.launchActivity(mIntentToStartActivity);
         /*
          * case 1: url = "http://login.microsoftonline.com/"
          * case 2: url = "https://login.microsoftonline.com/"
          */
-        String url = "https://login.microsoftonline.com/";
-        WebViewClient client = getCustomWebViewClient();
-        WebView mockview = new WebView(getActivity().getApplicationContext());
-        ReflectionUtils.setFieldValue(mActivity, "mSpinner", null);
-        assertEquals(false, client.shouldOverrideUrlLoading(mockview, url));
+        final String url = "https://login.microsoftonline.com/";
+        final WebViewClient client = getCustomWebViewClient();
```

Comment:
There appears to be some copy/paste looking code here - can we reduce this in some way?